tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1012,3 +1015,210 @@ def test_copy_object_during_pod_restart(self):
+        resp = self.ha_api.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=new_bucket, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[new_bucket])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        resp = self.s3_test_obj.get_object(new_bucket, self.object_name)
+        latest_vetag = self.version_etag[new_bucket][-1]
+        latest_v = list(latest_vetag.keys())[0]
+        etag = list(latest_vetag.values())[0]
+        if resp[1]["VersionId"] != latest_v:
+            assert_utils.assert_true(False, "Get Object without specifying VersionID does not "
+                                            f"match with latest {latest_v} {resp[1]}")
+        if resp[1]["ETag"] != etag:
+            assert_utils.assert_true(False, "Etag without specifying VersionID does not match with"
+                                            f" etag {etag} {resp[1]}")
+        LOGGER.info("Step 11: Got versions of %s with & without specifying VersionID & verified "
+                    "etag for %s", self.object_name, new_bucket)
+        # TODO: Need to check if enabling or disabling versioning is supported or not with dtm
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 12: Suspend versioning on %s.", new_bucket)
+            resp = self.s3_ver.put_bucket_versioning(bucket_name=new_bucket, status="Suspended")
+            assert_utils.assert_true(resp[0], resp)
+            LOGGER.info("Step 12: Suspended versioning on %s.", new_bucket)
+            LOGGER.info("Step 13: Upload same object %s after suspending versioning and verify its "
+                        "null for %s.", self.object_name, new_bucket)
+            args = {'file_path': download_path, 'chk_null_version': True, 'is_unversioned': True}
+            resp = self.ha_api.parallel_put_object(event, self.s3_test_obj, new_bucket,
+                                                   self.object_name, **args)
+            assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+            self.version_etag[new_bucket].extend(resp[1])
+            LOGGER.info("Step 13: Uploaded same object %s after suspending versioning and verified "
+                        "its null for %s.", self.object_name, new_bucket)
+
+            LOGGER.info("Step 14: Verify existing versions are remained intact")
+            for bucket in bucket_list:
+                resp = self.ha_api.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                                       bkt_name=bucket, obj_name=self.object_name,
+                                                       ver_etag=self.version_etag[bucket])
+                assert_utils.assert_true(resp[0], f"Get object with versionID failed {resp[1]} for"
+                                                  f" {bucket}")
```

Comment:
Don't we need to match previous version IDs with after restart version IDs?

Reply:
Yes. Same is taken care under step 11 with DTM flag disable setting.
This was taking care of previous version IDs created after shutdown. But now with dtm flag, we won't be able to create new bucket so not repeating the same.

Reply:
I don't think moving step under dtm condition was needed.
But verifying versions before shutdown and after restart is needed. Are we storing versions before shutdown somewhere so that those can be verified here?

Reply:
We are maintaining dict of all the version list associated with the bucket created before shutdown. And sending the same to compare with what will be fetched with call under step 9 for parallel_get_object

In case of dtm condition, bucket_name will have same bucket for which we have already checked versions in step 6. Didn't want to repeat that.